### PR TITLE
Fix App#renameRoom

### PR DESCRIPTION
**Pull request by TheJetOU**
It was missing some features like updating title bar, autojoins, etc.

Server PR: smogon/pokemon-showdown#6295.

### CODE REVIEW

**Comment by Zarel** [comment]
`roomid === toRoomid(title)` is not necessarily guaranteed.

For instance, `battle-ou-6234532` might have title `[Gen 7] OU: Zarel vs Yuyuko`

And `wifi` might have title `Wi-Fi`.

Neither of these match `toRoomid`.

**Comment by TheJetOU** [comment]
I suppose we need to make the server send `roomid|title`.

File: js/client.js [hunk 1]
```diff
@@ -1722,18 +1723,25 @@ function toId() {
 			if (room.requestLeave && !room.requestLeave(e)) return false;
 			return this.removeRoom(id);
 		},
-		renameRoom: function (id, newid) {
+		renameRoom: function (id, newtitle, newid) {
```
Comment (Zarel) [inline js/client.js]:
```suggestion
		renameRoom: function (id, newid, newtitle) {
```

Bikeshedding, but `newid` should be first for backwards compatibility reasons.

File: js/client.js [hunk 2]
```diff
@@ -1722,18 +1723,25 @@ function toId() {
...
+			var newid = newid || toRoomid(newtitle);
```
Comment (Zarel) [inline js/client.js]: And `|| toRoomid(newtitle)` should be dropped; it should instead be `newtitle = newtitle || newid`.

File: js/client.js [hunk 3]
```diff
@@ -879,7 +879,9 @@ function toId() {
 						self.send('/join ' + roomid);
 					});
 				} else if (data === 'rename') {
-					this.renameRoom(roomid, errormessage);
+					// |newtitle|newid
```
Comment (Zarel) [inline js/client.js]: This should also be `|newid|newtitle`.

Unless it wasn't in the past?

Comment (TheJetOU) [reply js/client.js]: > Unless it wasn't in the past?

It didn't exist in the past. I just appended roomid to it in the server code when I realized it was necessary. Is there a difference?

Comment (Zarel) [reply js/client.js]: Was it accepting titles in the past?

Comment (Zarel) [reply js/client.js]: It seems like `|rename|newid` was the old signature. If you want to support titles, it should go after the id, not before.

**Pull request merged**